test/test_optim.py

```diff
@@ -2205,69 +2206,66 @@ def test_non_empty_state(self, device, dtype, optim_info):
         dtypes=[torch.float32],
     )
     def test_less_mem_beta1_zero_adam(self, device, dtype, optim_info):
-        # test that beta1=0.0 uses less memory than the default
-        model = torch.nn.Linear(5, 5)
-        model.to(dtype=dtype, device=device)
-        inpt = torch.rand(2, 5, dtype=dtype, device=device)
-
-        optim_inputs = optim_info.optim_inputs_func(device=device)
-        
-        mem_used_exp_avg = 0 
-        case_to_mem_usage = {}
-        
         def get_obj_size(d):
             if isinstance(d, dict):
                 return sum(get_obj_size(v) for v in d.values())
-            elif isinstance(d, (list,tuple)):
+            elif isinstance(d, (list, tuple)):
                 return sum(get_obj_size(v) for v in d)
             elif isinstance(d, torch.Tensor):
                 return d.element_size() * d.numel()
             return d.__sizeof__()
-            
-        for optim_input in optim_inputs:
-            beta1_zero = (
-                "betas" in optim_input.kwargs and optim_input.kwargs["betas"][0] == 0.0
-            )
-            if beta1_zero or optim_input.desc == "default":
-                optim = optim_info.optim_cls(model.parameters(), **optim_input.kwargs)
-                optim.zero_grad() 
-                output = model(inpt)
-                loss = output.sum()
-                loss.backward()
-                optim.step()
 
-                if not beta1_zero : 
-                    mem_used_exp_avg += sum(
-                        optim.state[p]["exp_avg"].element_size()
-                        * optim.state[p]["exp_avg"].numel()
-                        for p in model.parameters()
-                    ) 
-                    
-                case_to_mem_usage[
-                    "beta1_zero" if beta1_zero else "default"
-                ] = get_obj_size(optim.state_dict()) 
-            
-                
-        self.assertGreaterEqual(
-            case_to_mem_usage["default"]-case_to_mem_usage["beta1_zero"], mem_used_exp_avg 
-        )
-        
+        # test that beta1=0.0 uses less memory than the default
+        beta_1_optim_inputs = [
+            o
+            for o in optim_info.optim_inputs_func(device=device)
+            if "betas" in o.kwargs and o.kwargs["betas"][0] > 0.0
+        ]
+
+        num_params = 4
+        print(f"num_params: {num_params}")
+        print(f"sizeof: {32 * 16 * dtype.__sizeof__()}")
+        size_of_param_in_bytes = (
```

Comment:
ah, you can also compute this later when you have the params, so like

num_params * params[0].nelement() * params[0].element_size()